.github/workflows/msvc.yml

```diff
@@ -0,0 +1,124 @@
+name: msvc
+
+on: [push]
+
+jobs:
+  compile:
+    name: ${{matrix.os}}-vc-${{matrix.VCPKG_PLATFORM_TOOLSET}}-${{matrix.CMAKE_GENERATOR_PLATFORM}}-${{matrix.BUILD_TYPE}}-${{matrix.BUILD_SHARED_LIBS}}
+
+    strategy:
+      matrix:
+        BUILD_TYPE: [Release, Debug]
+        BUILD_SHARED_LIBS: [OFF]
+        #VCPKG_PLATFORM_TOOLSET: [v143, v142, v141]
+        CMAKE_GENERATOR_PLATFORM: [x64, Win32]
+        os: [windows-latest]
+        include:
+          # MSVC 2022
+          - triplet: x64-windows
+            VCPKG_PLATFORM_TOOLSET: v143
+            CMAKE_GENERATOR_PLATFORM: x64
+
+          - triplet: x86-windows
+            VCPKG_PLATFORM_TOOLSET: v143
+            CMAKE_GENERATOR_PLATFORM: Win32
+
+          # MSVC 2019
+          - triplet: x86-windows
+            VCPKG_PLATFORM_TOOLSET: v142
+            CMAKE_GENERATOR_PLATFORM: Win32
+ 
+          # MSVC 2017
+          - triplet: x86-windows
+            VCPKG_PLATFORM_TOOLSET: v141
+            CMAKE_GENERATOR_PLATFORM: Win32
```

Comment:
Should we make x64 the default maybe?
Also (later) we should consider reducing amount of time the builds will run (too many variants)

Reply:
> Should we make x64 the default maybe? 

You can release x64 version.

> Also (later) we should consider reducing amount of time the builds will run (too many variants)

I think more attention should be paid to possible compilation errors.